src/algebraic_topology/alternating_face_map_complex.lean

```diff
@@ -0,0 +1,365 @@
+/-
+Copyright (c) 2021 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+
+import algebra.homology.homological_complex
+import algebraic_topology.simplicial_object
+import algebraic_topology.Moore_complex
+import category_theory.abelian.basic
+import algebra.big_operators.basic
+import tactic.ring_exp
+
+/-!
+
+# The alternating face map complex of a simplicial object in a preadditive category
+
+We construct the alternating face map complex, as a
+functor `alternating_face_map_complex : simplicial_object C ⥤ chain_complex C ℕ`
+for any preadditive category `C`. For any simplicial object `X` in `C`,
+this is the homological complex `... → X_2 → X_1 → X_0`
+where the differentials are alternate sums of faces.
+
+We also construct the natural transformation `inclusion_of_Moore_complex :
+nat_trans (normalized_Moore_complex A) (alternating_face_map_complex A)`
+when `A` is an abelian category
+
+## References
+* https://stacks.math.columbia.edu/tag/0194
+* https://ncatlab.org/nlab/show/Moore+complex
+
+-/
+
+open category_theory category_theory.limits category_theory.subobject
+open opposite
+open_locale big_operators
+
+noncomputable theory
+
+namespace algebraic_topology
+
+namespace alternating_face_map_complex
+
+/-- In degree n, the alternating face map complex is given by
+the nth-object of the simplicial object -/
+@[simp]
+def obj_X {C : Type*} [category C] (X : simplicial_object C) (n : ℕ) :=
+X.obj(op(simplex_category.mk n))
+
+variables {C : Type*} [category C] [preadditive C]
+variables (X : simplicial_object C)
+variables (Y : simplicial_object C)
+
+/-- The differential on the alternating face map complex is the alternate
+sum of the face maps -/
+@[simp]
+def obj_d (n : ℕ) : (obj_X X (n+1)) ⟶ (obj_X X n) :=
+∑ i in finset.range(n+2), ((-1 : ℤ)^i • X.δ i)
+
+/-!
+## Proof of the chain complex relation `d ≫ d`
+
+The expansion of `d ≫ d` involves a double sum, or a sum of terms
+indexed by a set of the form {0,...,n} × {0,...,n+1}. We shall show
+a general cancellation lemma `antisymmetric_sum_cancels` of such sums when
+the terms f_{i,j} satisfy an "antisymmetry" relation f_{i,j+1} = -f_{j,i}
```

Comment:
I think the proof of `d ≫ d = 0` can be streamlined a bit. See also https://github.com/leanprover-community/lean-liquid/blob/master/src/for_mathlib/simplicial/complex.lean#L29L86

Reply:
Thanks for the remarks. I have done the trivial changes. I will see what I can do for the core of the proof.

Reply:
I have done the changes in the proof (learnt a lot about fin(n) vs finset.range(n)...). Then, I am just wondering whether we should add Adam Topaz as author also because the basic structure of the Lean proof is his.

Reply:
Makes sense, I don't think he'll object.